src/tokenized_dataloader.py

```diff
@@ -0,0 +1,53 @@
+import json
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torch.nn.utils.rnn import pad_sequence
+
+import nltk
+from nltk.tokenize import word_tokenize
+
+class SarcasmDataset(Dataset):
+    def __init__(self, json_file):
+        # Load data from the JSON file
+        self.data = []
+        with open(json_file, 'r') as f:
+            for line in f:
+                self.data.append(json.loads(line.strip()))
+
+        # Get max token length for padding
+        max_length = 0
+        for item in self.data:
+            tokens = item['tokens']
+            if len(tokens) >= max_length: max_length = len(tokens)
+        self.max_length = max_length
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        # Get the individual sample
+        item = self.data[idx]
+        label = torch.tensor(item['is_sarcastic'])
+        tokens = item['tokens']
+        
+        # crop for max length (shouldnt be needed but just in case)
+        tokens = tokens[:self.max_length]
+        
+        # Pad the tokens to the max length if needed
+        padding_length = self.max_length - len(tokens)
+        tokens.extend(['<pad>'] * padding_length)  # Padding with '<pad>'
```

Comment:
I would suggest padding with the end of string token https://github.com/tsunami935/nlp-sarcasm-detection/blob/f3f8152d431769097ae87758c16c797499a09d86/src/common.py#L2.
"\<pad\>" token is not in the vocabulary.

Reply:
Also `extend` is slower than concatenation.